source/cpu/cpu.go

```diff
@@ -276,9 +282,38 @@ func discoverTopology() map[string]string {
 		features["hardware_multithreading"] = strconv.FormatBool(ht)
 	}
 
+	if socketCount, err := getCPUSocketCount(); err != nil {
+		klog.ErrorS(err, "failed to get sockets count")
+	} else {
+		features["socket_count"] = strconv.FormatInt(socketCount, 10)
+	}
+
 	return features
 }
 
+func getCPUSocketCount() (int64, error) {
+	files, err := os.ReadDir(hostpath.SysfsDir.Path("bus/cpu/devices"))
+	if err != nil {
+		return 0, err
+	}
+
+	uniquePhysicalIDs := make(map[int]bool)
```

Comment:
nit: maybe just `map[int]struct{}` because we don't really use the value for anything.

```suggestion
	uniquePhysicalIDs := make(map[int]struct{})
```

Alternatively, I think, we could start using the `k8s.io/apimachinery/pkg/util/sets` (e.g. `sets.NewInt()` or `sets.NewString()` in this case) package in NFD as we already depend on apimachinery.

Reply:
that makes more sense.